torch/csrc/jit/passes/onnx/shape_type_inference.cpp

```diff
@@ -1471,13 +1471,29 @@ void ComputeConstant(Node* n, int opset_version) {
         if (input0_shape_size.has_value()) {
           auto input0_shape_value = input0_shape_size.value();
           if (ConstantValueMap::HasValue(n->input(1)->debugName())) {
+            // When value of `shape` is statically known.
+            // Shape of expand output can be computed.
             auto shape_temp = ConstantValueMap::GetValueInto1DInt64Vector(
                 n->input(1)->debugName());
             auto final_shape =
                 ComputeShapeFromExpand(input0_shape_value, shape_temp);
             if (final_shape.has_value()) {
               UpdateShape(n->output(), final_shape.value());
             }
+          } else if (ConstantValueMap::HasShape(n->input(1)->debugName())) {
+            // When shape of `shape` is statically known.
+            // Rank of expand output can be computed.
+            auto expand_shape =
+                ConstantValueMap::GetShapeInto1DInt64VectorWithOneUnknown(
+                    n->input(1)->debugName());
+            if (expand_shape.has_value() && expand_shape.value().size() == 1 &&
```

Comment:
Does this condition mean `shape` has rank 1 and the size of dimension 0 is known?
If so, why do we need to check that condition?
Why can't we handle e.g. rank = 2?

Reply:
goal is to check dimension size is positive. `shape` input of expand is by spec 1-d tensor, hence rank 1. Checking is to avoid segfault when `expand_shape` comes up as unexpected. 

Moving first part of condition to assertion.